pkg/nanny/nanny_test.go

```diff
@@ -349,13 +359,72 @@ func TestNannyTimer(t *testing.T) {
 	}
 }
 
+func TestNannyAllClear(t *testing.T) {
```

Comment:
Nice that we have a test for this 👍 could you also write a test as how you'd expect this to behave in reality?
Something like:
```
program running normally
- signal
- signal
- signal
- no signal
- trigger notify
- all-clea signal
```

Reply:
I have modified the test accordingly.